src/robotlibcore.py

```diff
@@ -142,6 +135,37 @@ def get_keyword_documentation(self, name):
             doc = '{}\n\n{}'.format(doc, tags) if doc else tags
         return doc
 
+    def get_keyword_types(self, keyword_name):
+        method = self.keywords.get(keyword_name)
+        if not method:
+            raise ValueError('%s is not keyword.' % keyword_name)
+        robot_types = self._get_robot_types(method)
+        if robot_types is not False:
+            if robot_types is None:
+                return {}
+            return robot_types
+        return self._get_annotations(method)
+
+    def _get_robot_types(self, method):
+        if hasattr(method, 'robot_types'):
+            robot_types = method.robot_types
+            if robot_types == tuple():
+                return False
+            return robot_types
+
+    def _get_annotations(self, method):
```

Comment:
This function is named `_get_annotations` but it also reads info from default values. That's not good for two reasons:

- The function name is misleading. This is mainly a style issue.
- Defaults should be read _also_ when type info is got fro `robot_types`. This is a real bug.

Reply:
1) Yes, did overlook the name, will fix it
2) Good catch, it is also missing a test.